src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/UnstructuredAddress.java

```diff
@@ -40,12 +43,20 @@ public static UnstructuredAddress fromString(final String edifact) {
             System.arraycopy(splitByColon, 1, addressLines, 0, splitByColon.length - 1);
         }
 
-        final String postCode = splitByPlus[INDEX_POSTCODE];
+        final String postCode = arrayGetSafe(splitByPlus, INDEX_POSTCODE).orElse(null);
         final String qualifier = edifact.startsWith(KEY + PLUS_SEPARATOR + PLUS_SEPARATOR + FORMAT) ? FORMAT : "";
 
         return new UnstructuredAddress(qualifier, addressLines, postCode);
     }
 
+    public Optional<String[]> getAddressLines() {
```

Comment:
Is this the only place in whole EDIFACT that can have address? If not, amend other segments with address

Reply:
So, there some other places where you can have an address. However, there are a bit different.

1) HealthAuthorityNameAndAddress. Example: NAD+FHS+XX1:954'
2) GpNameAndAddress. Example: NAD+GP+2750922,295:900'
3) RequesterNameAndAddress. Example: NAD+PO+++NORTH DOWN GP'
4) PerformerNameAndAddress. Example: NAD+SLA+++ST JAMES?'S UNIVERSITY HOSPITAL'

Nothing similar to UnstructuredAddress